src/normalizer.ts

```diff
@@ -151,17 +151,57 @@ export const normalizeHexOrComplexColor = (
 
 export const normalizeText = (t: TextChild): InternalTextPart[] => {
   if (isReactElementOrElementArray(t)) {
-    return flattenChildren(t).map(
+    return flattenChildren(t).reduce<InternalTextPart[]>(
       (
+        textParts,
         el:
           | string
           | number
           | ReactElement<TextLinkProps>
           | ReactElement<TextBulletProps>
       ) => {
         if (React.isValidElement(el)) {
-          let link;
           let bullet;
+          if (isTextBullet(el)) {
+            // We know the intention is for a bullet, so pass on true if no customisation required
+            const { children, style, rtlMode, lang, ...bulletProps } = el.props;
+            bullet = Object.keys(bulletProps).length ? bulletProps : true;
+
+            // If the bullet has multiple TextChild components, then we make
+            // the first one take on `bullet` prop and collect the rest up.
+            let childParts: InternalTextPart[] = [];
+            if (Array.isArray(children)) {
+              childParts = children
+                .map((child) => normalizeText(child))
+                .reduce((prev, current) => prev.concat(current), []);
+            } else {
+              childParts = normalizeText(children);
+            }
+
+            const color = childParts[0].style?.color ?? style?.color;
+            // Make `breakLine = false` for all child components except the last one
+            // (so every child will sit within the same bullet point)
+            const firstChild = {
+              rtlMode,
+              lang,
+              ...childParts[0],
+              bullet,
+              breakLine: childParts.length <= 1,
+              style: {
+                ...(style || {}),
```

Comment:
Does not having this inside the `map` below mean that `style` contents will only be copied to the first bullet part? I'm not sure if this would matter

Reply:
Good point! For some reason I thought that these styles would implicitly apply to all the "child nodes" if they're missing the styles like with nesting `<Text>` elements inside other `<Text>` elements; but actually the way I've implemented this as side-by-side text items wouldn't allow for that. The most reasonable expectation is that parent node styles would apply to child nodes, so this would be a good addition.